src/components/frame/Extensions/pages/MultiSigPage/utils.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  type DomainThresholdConfig,
+  type MultiSigSetupFormValues,
+  MultiSigThresholdType,
+} from './types.ts';
+
+export const getThresholdType = (value?: number) =>
+  value && value > 0
+    ? MultiSigThresholdType.FIXED_THRESHOLD
+    : MultiSigThresholdType.MAJORITY_APPROVAL;
+
+export const getInitialGlobalThreshold = (multiSigConfig) =>
+  (multiSigConfig ? multiSigConfig.colonyThreshold : 0) || '';
+
+export const getInitialDomainConfig = (domain, multiSigConfig) => {
+  const { colonyThreshold } = multiSigConfig;
+
+  const existingThreshold = multiSigConfig.domainThresholds?.find((item) => {
+    return Number(item?.domainId) === domain.nativeId;
+  })?.domainThreshold;
+  let type = MultiSigThresholdType.INHERIT_FROM_COLONY;
+
+  if (existingThreshold != null && existingThreshold !== colonyThreshold) {
+    type = getThresholdType(existingThreshold);
+  }
+
+  const name = domain.metadata?.name || '';
+
+  return {
+    id: domain.id,
+    nativeSkillId: Number(domain.nativeSkillId),
+    type,
+    name,
+    threshold: existingThreshold || colonyThreshold || 0,
+  };
+};
+
+export const getDomainThresholds = (
+  values: MultiSigSetupFormValues,
+  domainThresholdConfigs: DomainThresholdConfig[],
+  thresholdType: MultiSigThresholdType,
+) =>
+  domainThresholdConfigs.map((domain) => {
+    let threshold = 0;
```

Comment:
I'd drop a comment saying we default to `INHERIT_FROM_COLONY` just to reduce potential head scratches in the future :D